[safe-merge-queue] Make check-status-of-pr step prettier

briannafan opened the pull request:
#### 4235d273c502b6aefad13fd572748aae1737d902
<pre>
[safe-merge-queue] Make check-status-of-pr step prettier
<a href="https://bugs.webkit.org/show_bug.cgi?id=263593">https://bugs.webkit.org/show_bug.cgi?id=263593</a>
rdar://problem/117505427

Reviewed by Jonathan Bedard.

Condense EWS status check outputs and mark step as success if still pending.

* Tools/CISupport/ews-build/steps.py:
(CheckStatusOfPR):
(CheckStatusOfPR.run):
(CheckStatusOfPR.getResultSummary):
(CheckStatusOfPR.getQueueStatusFromList):

Canonical link: <a href="https://commits.webkit.org/269985@main">https://commits.webkit.org/269985@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b4b6879e634c3b66f1a3f19a1a80bd37918fc4eb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/23756 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/1868 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/24858 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/25905 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21914 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/24030 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3453 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24267 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22449 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/23999 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1422 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20544 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26500 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1190 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21451 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27709 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21664 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21726 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25486 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1152 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18848 "Passed tests") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/23544 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/1144 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5793 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1564 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1459 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -2521,7 +2522,7 @@ def getQueueStatusFromList(self, change_id, pr_number):
             yield self._addToLog('stdio', 'Required checks are not completed. Waiting until all checks are completed before relabelling PR.\n')
             yield self._addToLog('stdio', f'Missing the following checks: {missing_checks}\n')
-            defer.returnValue(False)
+            defer.returnValue(2)
```
Can you name these variables something? It's hard to understand what you expect them to mean (since it's clearly not a normal program status code)

JonWBedard commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -2495,22 +2497,21 @@ def getQueueStatusFromList(self, change_id, pr_number):
                 else:  # null
                     missing_checks.append(queue)
-                    yield self._addToLog('stdio', f'Pending\n')
+                    yield self._addToLog('stdio', f'{queue}: Pending\n')
```
Can we differentiate "pending" and "skipped?"

JonWBedard commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -2424,6 +2424,10 @@ class CheckStatusOfPR(buildstep.BuildStep, GitHubMixin, AddToLogMixin):
+    EWS_WEBKIT_FAILED = 0
+    EWS_WEBKIT_PASSED = 1
+    EWS_WEBKIT_PENDING = 2
+    passed_status_check = None
```
This should be part of the `__init__`, not a class variable.

briannafan committed:
[safe-merge-queue] Make check-status-of-pr step prettier
https://bugs.webkit.org/show_bug.cgi?id=263593
rdar://problem/117505427

Reviewed by Jonathan Bedard.

Condense EWS status check outputs and mark step as success if still pending.

* Tools/CISupport/ews-build/steps.py:
(CheckStatusOfPR):
(CheckStatusOfPR.run):
(CheckStatusOfPR.getResultSummary):
(CheckStatusOfPR.getQueueStatusFromList):

Canonical link: https://commits.webkit.org/269985@main
